ingestr/src/sources.py

```diff
@@ -997,6 +997,77 @@ def dlt_source(self, uri: str, table: str, **kwargs):
         ).with_resources(endpoint)
 
 
+class TikTokSource:
+    # tittok://?access_token=<access_token>&advertiser_id=<advertiser_id>
+    def handles_incrementality(self) -> bool:
+        return True
+
+    def dlt_source(self, uri: str, table: str, **kwargs):
+        endpoint = "custom_reports"
+
+        parsed_uri = urlparse(uri)
+        source_fields = parse_qs(parsed_uri.query)
+
+        access_token = source_fields.get("access_token")
+        if not access_token:
+            raise ValueError("access_token is required to connect to TikTok")
+
+        time_zone = source_fields.get("time_zone", "UTC")
+
+        advertiser_id = source_fields.get("advertiser_id")
+        if not advertiser_id:
+            raise ValueError("advertiser_id is required to connect to TikTok")
+
+        if kwargs.get("interval_start"):
+            start_date = ensure_pendulum_datetime(
+                str(kwargs.get("interval_start"))
+            ).in_tz(time_zone[0])
+        else:
+            Default_date = pendulum.now().subtract(days=90)
```

Comment:
Do we need a default start date in order to be able to load data?

Reply:
Yes, we need it in this case because the API expects a start and end date.